### DISCUSSION TITLE
How to get a user's "preferred" profile URL?

### DISCUSSION THREAD

**Discussion by snarfed** [bluesky-social/atproto#1652] [How to do it?]
Right now, bsky.app is the official web client that more or less everyone uses. In the future, hopefully, that may be less true. If I have a user's DID (or handle), do you all have a way in mind that I could get their "preferred" web profile URL?

For now, I can hard-code bsky.app profile URLs, but I'm wondering if there will be a better way in the future.

...or am I thinking about this wrong, and choice of client should be up to the viewing user, in the spirit of "user agent," not the user being viewed? Which would make sense...except then we still wouldn't have guidance for people making non-client or non-user-facing tools that need to expose _some_ web profile URL for a given user. Can those kinds of tools do anything better than hard coding bsky.app?

Thanks in advance!

**Comment by bnewbold** [comment]
It's a good question! Don't want to over-centralize on a single party's domain like `bsky.app`, but also nice to have something easy to recognize and parse.

A related issue, while talking about "canonical" URLs is that having the DID in there (instead of just handle) is nice in some situations.

A couple notes/ideas:

- fediverse/mastodon has some usability issues with many interfaces to the same content, when the interface looks very similar to the app you are logged in to, but you can't interact. there are probably ways to design around this?
- in the long long run maybe AT URIs will be more natively supported, eg in web (HTML and browsers), and you can just have them in `href`! would open in preferred user agent
- one nice simple option is to just go to the account's handle. eg `https://bnewbold.bsky.team/post/abc123` or similar as a post URL. could break on handle change, which is a bit of a bummber. but, depending on the handle config, under the account's control, and they can decide which app info to display on the home page, etc. maybe a semi-standard set of path shapes to link directly to Lexicon-specific content like a specific post/thread. config (CSS? MIDI tracks?) could be a record in-repo.
- I think it would be great and probably doable to have a neutral and generic (not app-specific) repo explorer interface for debugging and looking at records. like `atproto.data/<did>/<nsid>/<rkey>` or something. and this service could have some basic logic to display repo-specific or Lexicon-specific links out to more canonical websites
- i'm not sure what the current OS/platform support for this is these days, but potentially an app-specific URL like `bsky.app`, which is associated with the `app.bsky.*` Lexicons, could be registered as a URI-like prefix. eg, you could install Greysky (or some other client), and when you click an https:// URL with that hostname it gets opened with the registered app, with browser as a fall-back
